Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm

```diff
@@ -162,48 +163,86 @@ static void cleanUpTestWebPushD(NSURL *tempDir)
     EXPECT_NULL(error);
 }
 
-template <typename T>
-static void addMessageHeaders(xpc_object_t request, T messageType, uint64_t version)
-{
-    xpc_dictionary_set_uint64(request, WebKit::WebPushD::protocolMessageTypeKey, static_cast<uint64_t>(messageType));
-    xpc_dictionary_set_uint64(request, WebKit::WebPushD::protocolVersionKey, version);
-}
+class WebPushXPCConnectionMessageSender : public IPC::MessageSender {
+public:
+    WebPushXPCConnectionMessageSender(xpc_connection_t connection)
+        : m_connection(connection)
+    {
+    }
+    ~WebPushXPCConnectionMessageSender() final { };
 
-void sendMessageToDaemon(xpc_connection_t connection, MessageType messageType, const Vector<uint8_t>& message, uint64_t version = WebKit::WebPushD::protocolVersionValue)
-{
-    auto request = adoptNS(xpc_dictionary_create(nullptr, nullptr, 0));
-    addMessageHeaders(request.get(), messageType, version);
-    xpc_dictionary_set_data(request.get(), WebKit::WebPushD::protocolEncodedMessageKey, message.data(), message.size());
-    xpc_connection_send_message(connection, request.get());
-}
+    void incrementProtocolVersionForTesting() { m_incrementProtocolVersionForTesting = true; }
 
-xpc_object_t sendMessageToDaemonWithReplySync(xpc_connection_t connection, MessageType messageType, const Vector<uint8_t>& message, uint64_t version = WebKit::WebPushD::protocolVersionValue)
+private:
+    bool performSendWithoutUsingIPCConnection(UniqueRef<IPC::Encoder>&&) const final;
+    bool performSendWithAsyncReplyWithoutUsingIPCConnection(UniqueRef<IPC::Encoder>&&, CompletionHandler<void(IPC::Decoder*)>&&) const final;
+
+    OSObjectPtr<xpc_object_t> messageDictionaryFromEncoder(UniqueRef<IPC::Encoder>&&) const;
+
+    IPC::Connection* messageSenderConnection() const final { return nullptr; }
+    uint64_t messageSenderDestinationID() const final { return 0; }
+    OSObjectPtr<xpc_connection_t> m_connection;
```

Comment:
It'd be nice to have a blank line between to separate data members from member functions.